controller/api/models.py

```diff
@@ -287,8 +288,9 @@ def _start_containers(self, to_add):
         [t.start() for t in create_threads]
         [t.join() for t in create_threads]
         if any(c.state != 'created' for c in to_add):
-            err = 'aborting, failed to create some containers'
+            err = 'aborting, failed to create some containers rolling back'
             log_event(self, err, logging.ERROR)
+            self._destroy_containers(to_add)
```

Comment:
we already make this call [here](https://github.com/deis/deis/blob/master/controller/api/views.py#L193-L195) which will destroy the containers, though I agree that it really should be here. Would you mind updating the logic I just pointed out so we don't call `_destroy_containers()` twice?